Fix inconsistent icons in the app toolbar.

flofriday opened the pull request:
### Description
**Closes:** #7128
The problem was that the icons have their own color they use (`attr/action_icon_color`) which is fine, except in the toolbar where they should use a different color.

The fix applies a style to all toolbars where in the style only adds a theme for the surface which remaps the `attr/action_icon_color` to the color the other icons in the toolbar use.

### Checklist
<!-- 
  To help us keep the issue tracker clean and work as efficient as possible,
  please make sure that you have done all of the following.
  You can tick the boxes below by placing an x inside the brackets like this: [x]
-->
- [x] I have read the contribution guidelines: https://github.com/AntennaPod/AntennaPod/blob/develop/CONTRIBUTING.md#submit-a-pull-request
- [x] I have performed a self-review of my code
- [x] I have run the automated code checks using `./gradlew checkstyle spotbugsPlayDebug spotbugsDebug :app:lintPlayDebug`
- [x] My code follows the style guidelines of the AntennaPod project: https://github.com/AntennaPod/AntennaPod/wiki/Code-style
- [x] I have mentioned the corresponding issue and the relevant keyword (e.g., "Closes: #xy") in the description (see https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [x] If it is a core feature, I have added automated tests

ByteHamster commented:
Hmm, I think this looks a bit weird with the black title and gray icons. It also seems that many other apps with white toolbars (WhatsApp, Signal, Firefox, Tasks, Material Files, even the Android Settings app) use full-black action icons. I think we should switch the overflow menu icon to black instead of switching all other icons to gray.

flofriday commented:
No problem, now that I know where the right knobs are it should be super easy to switch to that ;)

flofriday commented:
I made them now vibrant.

I also discovered an issue with my solution which required me to remove the `android:theme="?attr/actionBarTheme"` from some screens. Removing that didn't have any visual impact and made my fix work on all screens. But maybe I don't know enough here and there was a good reason this was there.

Moreover I also discovered that the fix currently doesn't apply to the preferences, but I will look into that now.

flofriday commented:
I can't seem to figure out how to style the toolbar from the settings screen. It is a little bit difficult since we don't create it directly but `PreferenceScreen` creates it for us (at least I think that is what's happening).

ByteHamster commented:
> which required me to remove the android:theme="?attr/actionBarTheme" from some screens

This means `style name="Widget.AntennaPod.ActionBar"` is no longer applied to all toolbars. Not sure if that's a problem or if the style was just there for legacy reasons. The "blame" button on GitHub (or `git blame`) might give a cue why this style was added in the first place. We should also do some testing on devices/emulators with different Android versions and with/without the "dynamic colors" checkbox ticked.

I think we should be consistent across all screens, either removing the style completely or keeping it everywhere.

flofriday commented:
I agree, about consitency. Tracking down the changes I found that in commit 7f4d43deb16ee0ce6a49f78805e5600d3e01161f you refactored the toolbar and added the theme as it wasn't there before. 

However later we switched from `androidx.appcompat.widget.Toolbar` to `com.google.android.material.appbar.MaterialToolbar` in commit ac8114342c4beb90745cfc5bd9235ca219a36b29.
So maybe it was only needed for the old toolbar and the new one doesn't neeed it anymore.

Anyway, I will do some testing on older devices but so far I am leaning towards removing it everywhere.

flofriday commented:
I just tested it on Android 5 (API 21) which is our lowest android target and the app still looks great. There are very view screens with an ugly toolbar but they are also there in on `develop` so my changes don't seem to destroy anything. 🎉 

**Screenshot of my changes:**
![image](https://github.com/AntennaPod/AntennaPod/assets/21206831/bc5ae0ab-1824-484b-83f4-d5346802cad5)


**Screenshot of the weired toolbar:**
Again, this is also on develop and since android 5 is so old and the screen is bareley used I don't think we should put too much effort into it.
<img width="490" alt="Screenshot 2024-05-06 at 11 59 23" src="https://github.com/AntennaPod/AntennaPod/assets/21206831/3178e79e-c0f4-4375-b68c-7b86c1cdd692">

ByteHamster commented:
Thanks!

I sneaked in a fix for the toolbar having a weird shadow :)

flofriday commented:
@ByteHamster your fix also makes the toolbar in the settings page blue with elevation, which looks kinda odd as the statusbar is still white.

![Screenshot_20240507_085508](https://github.com/AntennaPod/AntennaPod/assets/21206831/56326dcf-b174-4c0d-b770-983f65107732)

ByteHamster commented:
Oops, thanks. Apparently I didn't test well enough. Fixed in #7169
